### PR TITLE
To Do list: interactive list

### PR DESCRIPTION
**Pull request by calebchris000**
## The following features were added:

- Add a new JavaScript file and import it as a module:
- Add event listener to the checkbox (change).
- Update items object's value for completed key upon user actions.
- Implement a function for the "Clear all completed" button (use filter() method).
- Store the updated array of items in local storage, so the user gets the correct list of values after the page reloads (which means that any changes made to the list should be preserved).

### CODE REVIEW

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @calebchris000,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- GitHub flow was correctly used.  ✔️   
- Descriptive pull request.  ✔️ 
- It has one new JavaScript files imported as a module.  ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

- [ ] Hello @calebchris000, Everything seems fine so far, however, I was unable to run your application from my local computer, even after following the instructions in your README file. Instead I get this error message 👇🏻

image

- 🤔 I think the reason is that I am missing the necessary configuration in the package.json file to run your application. I don't think you should ignore the package JSON file, as that is where all the app project dependencies stay. if you can't provide all the information needed to run the application from the README, I can suggest removing the package.json file from your ignore file so we can easily run `npm install`, `npm run build` , and `npm start` to preview the application from the browser 💯 👍🏻 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by calebchris000** [comment]
@BenMukebo Got it. Thanks.

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @calebchris000, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️


 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Hello @calebchris000, Great job so far but accroding to the project requiements, The updated array of books should be stored in local storage, so the user gets the correct list values after the page reloads. However, when I reload the page, all the books are deleted except the first one,, Here is the [VIDEO](https://www.loom.com/share/0ab219c082a64f9183d0c520e16080da) link. please fix this problem and make sure you have correctly implemented the A function for the "Clear all completed" button using `Array.filter()`.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/clear.js [hunk 1]
```diff
@@ -8,12 +8,12 @@ export const clearComplete = () => {
 
   wrapperChecks.forEach((item) => {
     item.remove();
+  });
 
-    const get = JSON.parse(localStorage.getItem('data'));
+  const get = JSON.parse(localStorage.getItem('data'));
```
Comment (jasonfelice) [inline src/clear.js]: The local storage is working fine, however, there's an issue. When I clear all the completed tasks and refresh the page it renders an `undefined` task. Please refer to the attached image.

- [ ] Kindly fix this bug. Consider this: when the page loads, check if there's something on local storage using a conditional.

image

**Review by jasonfelice — CHANGES_REQUESTED** [review]
Hi @calebchris000, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
No linter errors✔️
Followed GitHub flow✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/clear.js (hunk 1 above)
Comment (calebchris000) [reply src/clear.js]: Okay, thank you for the correction.

File: src/clear.js [hunk 2]
```diff
@@ -1,19 +1,2 @@
...
-
-  const get = JSON.parse(localStorage.getItem('data'));
-
-  const cleaned = get.filter((x) => x.checked !== true);
-  newSave.add(cleaned);
-};
-
-export default clearComplete;
+// import { load } from "./generateList.js";
+// const todo = document.querySelector(".todo-list");
```
Comment (Felix45) [inline src/clear.js:2]: - [ Optional ] Please consider removing any pieces of commented code from your codebase, commented code makes your repository look unprofessional. Kindly rectify this.

File: src/generateList.js [hunk 3]
```diff
@@ -98,12 +98,31 @@ const refresh = () => {
...
+  }
+};
+
+export const clearComplete = () => {
+  const wrapperChecks = todo.querySelectorAll('.wrapperCheck');
+
+  wrapperChecks.forEach((item) => {
+    item.remove();
+  });
+  // wrapperChecks.length === 0 ?localStorage.clear():0
```
Comment (Felix45) [inline src/generateList.js]: - [ Optional ] Please consider removing any pieces of commented code from your codebase, commented code makes your repository look unprofessional. Kindly rectify this.

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @calebchris000 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] No linter errors :white_check_mark: 
- [x] GitHub flow is used correctly :white_check_mark:
- [x] Descriptive pull request title and summary :white_check_mark:  

## Optional suggestions
- [ Optional ] I've noticed that the License link on your readme file does not lead to the License in the project root directory, please consider changing the License Link to the one below to fix this. A working License link informs someone who would like to use your code or project how they are Licensed.

```
This project is [MIT](./LICENSE) licensed.
```
- [ Optional ] Please consider removing the following empty source files from your project since they are not being used anywhere in the project, an empty source file makes your repository unprofessional.
   
   - src/menuItem.js
   - src/refresh.js

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**